### DISCUSSION TITLE
How to render an animation of mesh, or with a draggable realtime GUI?

### DISCUSSION THREAD

**Discussion by SuX97** [facebookresearch/pytorch3d#781] [Q&A]
Hi,
I have read the docs about rendering a mesh at various views. I am wondering how can I render an animation of mesh, i.e. I change the mesh using blendshapes at each frame and render it into a video. Meanwhile, Is it possible to enable a GUI that is draggable like openGL for visualization of rendering instead of show or store using matplotlib?

Thanks!

**Comment by PerryXDeng** [comment]
For the GUI part, it is possible to use the MoviePy library that uses Matplotlib to render a series of images into a video. I am likewise curious about the ability to render a mesh with a batch of vertex position tables for animation. Currently, doing so will cause the renderer to only render the first vertex position table (the first frame).

**Comment by SuX97** [comment]
> For the GUI part, it is possible to use the MoviePy library that uses Matplotlib to render a series of images into a video. I am likewise curious about the ability to render a mesh with a batch of vertex position tables for animation. Currently, doing so will cause the renderer to only render the first vertex position table (the first frame).

My current walkaround is to render each frame and combine them using FFmpeg, while I still need a real-time rendering GUI, from which I can change the 3D assets(e.g. *.obj files) and observe the change of the animation.
